nf_core/pipelines/lint/pipeline_if_empty_null.py

```diff
@@ -0,0 +1,58 @@
+import fnmatch
+import logging
+import os
+import re
+from pathlib import Path
+
+log = logging.getLogger(__name__)
+
+
+def pipeline_if_empty_null(self, root_dir=None):
+    """Check for ifEmpty(null)
+
+    There are two general cases for workflows to use the channel operator `ifEmpty`:
+        1. `ifEmpty( [ ] )` to ensure a process executes, for example when an input file is optional (although this can be replaced by `toList()`).
+        2. When a channel should not be empty and throws an error `ifEmpty { error ... }`, e.g. reading from an empty samplesheet.
+
+    There are multiple examples of workflows that inject null objects into channels using `ifEmpty(null)`, which can cause unhandled null pointer exceptions.
+    This lint test throws warnings for those instances.
+    """
+    passed = []
+    warned = []
+    file_paths = []
+    pattern = re.compile(r"ifEmpty\(\s*null\s*\)")
+
+    # Pipelines don't provide a path, so use the workflow path.
+    # Modules run this function twice and provide a string path
+    if root_dir is None:
+        root_dir = self.wf_path
+
+    ignore = [".git"]
+    if Path(root_dir, ".gitignore").is_file():
+        with open(Path(root_dir, ".gitignore"), encoding="latin1") as fh:
+            for line in fh:
+                ignore.append(Path(line.strip().rstrip("/")).name)
+    for root, dirs, files in os.walk(root_dir, topdown=True):
+        # Ignore files
+        for i_base in ignore:
+            i = str(Path(root, i_base))
+            dirs[:] = [d for d in dirs if not fnmatch.fnmatch(str(Path(root, d)), i)]
+            files[:] = [f for f in files if not fnmatch.fnmatch(str(Path(root, f)), i)]
+        for fname in files:
+            try:
+                with open(Path(root, fname), encoding="latin1") as fh:
```

Comment:
I don't think we have latin1 enconding
```suggestion
                with open(Path(root, fname)) as fh:
```

Reply:
interestingly without either `encoding="latin1"` or `errors="ignore"` it fails to decode files both during the pytest and when testing the linting on a new pipeline.
I took this from the file [pipelines/lint/pipeline_todos.py](https://github.com/nf-core/tools/blob/main/nf_core/pipelines/lint/pipeline_todos.py). Is this a me issue or should I add `latin1` back in?

Reply:
actually I just realized [template_strings.py](https://github.com/nf-core/tools/blob/34bc33802e8b51cadb62a3f0c615cf7a947100f6/nf_core/pipelines/lint/template_strings.py#L56) uses `encoding="latin1"` for reading files too, I'll add it back in for now

Reply:
my bad, let's add `encoding="latin1"` back, thanks!